features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp

```diff
@@ -506,6 +506,43 @@ ble_error_t GenericSecurityManager::requestAuthentication(connection_handle_t co
 // MITM
 //
 
+ble_error_t GenericSecurityManager::generateOOB(
+    const address_t *address
+) {
+    /* legacy pairing */
+    ble_error_t status = get_random_data(_oob_temporary_key.buffer(), 16);
+
+    if (status == BLE_ERROR_NONE) {
+        _oob_temporary_key_creator_address = *address;
+
+        eventHandler->legacyPairingOobGenerated(
+            &_oob_temporary_key_creator_address,
+            &_oob_temporary_key
+        );
+    } else {
+        return status;
+    }
+
+    /* Secure connections. Avoid generating if we're already waiting for it.
+     * If a local random is set to 0 it means we're already calculating. */
+    if (!_oob_local_random.is_all_zeros()) {
```

Comment:
First time generation no longer a special case, any time it's zero it means we're waiting, zero is an illegal value for random. My rationale is that if zero is returned the stack is broken so no point asking it again, if it never returns - same, so this fails gracefully. Alternative is to simply have an extra bool but bytes are precious...

Reply:
on top of that it is safe to return even invalid random value to the stack because it will be zeroed since it was never delivered